csrc/java_evp_keys.cpp

```diff
@@ -328,6 +328,42 @@ JNIEXPORT jbyteArray JNICALL Java_com_amazon_corretto_crypto_provider_EvpEcPriva
     }
 }
 
+/*
+ * Class:     com_amazon_corretto_crypto_provider_EvpEdPrivateKey
+ * Method:    getPrivateKey
+ */
+JNIEXPORT jbyteArray JNICALL Java_com_amazon_corretto_crypto_provider_EvpEdPrivateKey_getPrivateKey(
+    JNIEnv* pEnv, jclass, jlong keyHandle)
+{
+    jbyteArray result = NULL;
+
+    try {
+        raii_env env(pEnv);
+
+        EVP_PKEY* key = reinterpret_cast<EVP_PKEY*>(keyHandle);
+
+        EVP_PKEY_CTX_auto ctx = EVP_PKEY_CTX_auto::from(EVP_PKEY_CTX_new(key, nullptr));
+        CHECK_OPENSSL(ctx.isInitialized());
+
+        OPENSSL_buffer_auto privateKeyBuffer;
+        PKCS8_PRIV_KEY_INFO_auto pkcs8 = PKCS8_PRIV_KEY_INFO_auto::from(EVP_PKEY2PKCS8(key));
+        CHECK_OPENSSL(pkcs8.isInitialized());
+
+        // This next line allocates memory
+        int bufInt = i2d_PKCS8_PRIV_KEY_INFO(pkcs8, &privateKeyBuffer);
```

Comment:
Check out `EVP_PKEY_get_raw_private_key()`. Simpler way to get the length.

Reply:
Another method: This function is type specific. So, you can just make the assumption you are taking an Ed25519 private key which has a fixed length.

I guess getting the size at runtime is slightly more memory safe (aws-lc does check the encoded size in the third argument though).

Reply:
Updated to use `EVP_PKEY_get_raw_private_key` to get bufSize